src/main/java/org/graylog/integrations/aws/codecs/AWSMetaCodec.java

```diff
@@ -0,0 +1,59 @@
+package org.graylog.integrations.aws.codecs;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.inject.assistedinject.Assisted;
+import org.graylog.integrations.aws.cloudwatch.KinesisLogEntry;
+import org.graylog2.plugin.Message;
+import org.graylog2.plugin.configuration.Configuration;
+import org.graylog2.plugin.configuration.ConfigurationRequest;
+import org.graylog2.plugin.inputs.annotations.ConfigClass;
+import org.graylog2.plugin.inputs.annotations.FactoryClass;
+import org.graylog2.plugin.inputs.codecs.AbstractCodec;
+import org.graylog2.plugin.inputs.codecs.Codec;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import javax.inject.Inject;
+
+public class AWSMetaCodec extends KinesisLogDataCodec {
```

Comment:
Should we name this `AWSKinesisMetaCodec`? It extends `KinesisLogDataCodec` so it's not really a generic AWS meta codec, right?

Reply:
Thanks for noticing this. I've removed the `KinesisLogDataCodec` extension. This was actually intended to be be a general AWS codec since there will be one AWSInput.

`KinesisLogDataCodec` has been renamed to `AbstractKinesisCodec` (the name had always confused me, even in the older AWS plugin). This class is only used for the two actual Kinesis codecs `KinesisRawLogCodec` and `KinesisCloudWatchFlowLogCodec`.